### PR TITLE
Add agancy->agency and friends

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -892,6 +892,8 @@ againt->against
 againts->against
 agaisnt->against
 agaist->against
+agancies->agencies
+agancy->agency
```
Comment (lurch) [inline codespell_lib/data/dictionary.txt:896]: `agant->agent` ?

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -892,8 +892,11 @@ againt->against
...
 aganda->agenda
 aganist->against
+agant->agent
```
Comment (lurch) [inline codespell_lib/data/dictionary.txt:899]: ...and the plural? :wink: (I kinda automatically assumed you'd be adding that too!)
Although perhaps it should be `agants->agents, against, giants,` ? :man_shrugging:

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -27423,6 +27426,11 @@ trabsform->transform
 traceablity->traceability
 trackign->tracking
 trackling->tracking
+tracsode->transcode
+tracsoded->transcoded
+tracsoder->transcoder
```
Comment (lurch) [inline codespell_lib/data/dictionary.txt:27431]: Just out of curiosity, how is this PR passing the CI checks?
http://app.aspell.net/lookup?dict=en_US-large;words=transcode%0D%0Atranscoded%0D%0Atranscoder

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -892,8 +892,11 @@ againt->against
 againts->against
 agaisnt->against
 agaist->against
+agancies->agencies
+agancy->agency
 aganda->agenda
```
Comment (lurch) [inline codespell_lib/data/dictionary.txt:897]: I wonder if `aganda` should also suggest `Uganda` ? :shrug:

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (sebweb3r) [reply codespell_lib/data/dictionary.txt:27431]: dictionary.txt is not checked against aspell.

Comment (lurch) [reply codespell_lib/data/dictionary.txt:27431]: LOL, shows what a n00b I am :wink: 
In that case, it's probably worth adding `tracsodes->transcodes` too.

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:896]: Done.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:910]: You should have suggested it then! :stuck_out_tongue: 

I've added against as a miss and a swap, but the a/i position and keyboard distance seems too great a stretch IMHO.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:27993]: Yeah sadly there are too many words not in their dictionary to guarantee everything is present.

Added that.

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:908]: It's another pretty big stretch, and not one of mine, but added anyway.

File: codespell_lib/data/dictionary_code.txt [hunk 5]
```diff
@@ -13,6 +13,7 @@ dont->don't
 dur->due
 endcode->encode
 errorstring->error string
+ether->either
```
Comment (larsoner) [inline codespell_lib/data/dictionary_code.txt:16]: -1 on this one, `ether` is a chemistry term AFAIK (i.e., used outside of code / just rare)

Comment (peternewman) [reply codespell_lib/data/dictionary_code.txt]: Okay I've dropped it entirely now. It was in rare but the first 7 pages of grep.app at least are genuine code usages, so I don't think it's actually very rare really.

File: codespell_lib/tests/test_dictionary.py [hunk 6]
```diff
@@ -189,6 +189,7 @@ def test_error_checking_in_aspell(err, rep, err_aspell, rep_aspell, match):
 # allow some duplicates, like "m-i-n-i-m-i-s-e", or "c-a-l-c-u-l-a-t-a-b-l-e"
 allowed_dups = {
     ('dictionary.txt', 'dictionary_en-GB_to_en-US.txt'),
+    ('dictionary.txt', 'dictionary_code.txt'),
```
Comment (larsoner) [inline codespell_lib/tests/test_dictionary.py:192]: No longer needed, right?

Comment (peternewman) [reply codespell_lib/tests/test_dictionary.py:192]: Probably not for the moment, although may appear again

**Pull request merged**